mewarpx/mewarpx/diags_store/field_diagnostic.py

```diff
@@ -58,9 +69,12 @@ def add_field_diag(self):
 
         mwxrun.simulation.add_diagnostic(diagnostic)
 
-    def add_plot_callback(self):
+    def add_diag_plot_callback(self):
         callbacks.installafterstep(self.plot_parameters)
 
+    def add_post_processing_check_callback(self):
+        callbacks.installafterstep(self.check_for_end_of_sim)
+
```

Comment:
I'm not sure we need to define separate functions in the class here, can we just install the callbacks directly from init? I don't see an advantage to the extra layer.